qiskit_nature/algorithms/excited_states_solvers/excited_states_eigensolver.py

```diff
@@ -58,17 +60,20 @@ def solver(self, solver: Union[Eigensolver, EigensolverFactory]) -> None:
     def solve(
         self,
         problem: BaseProblem,
-        aux_operators: Optional[List[Union[SecondQuantizedOp, PauliSumOp]]] = None,
+        aux_operators: Optional[ListOrDictType[Union[SecondQuantizedOp, PauliSumOp]]] = None,
+        dict_based_aux_ops: bool = False,
```

Comment:
I had to think about this flag a bit.

Its easy enough to see on the ElectronicStructureProblem, where there is a method to return a list or dictionary that we return a list as per the user expects since there are users calling that directly, using the top half of the stack so to speak, to get operators from a molecule which they then pass onto VQE.

If the user passes in a List of aux ops here shouldn't we just handle it as a list all the way down and back up the stack. This would be compatible with what is there now. If they pass in a dict, would be new code, then results would be in dict form too. I guess the flag comes into play if the user does not themselves supply any additional aux ops in that we can state things will become a dict in the future and this flag allows you to transition your code in that direction already by setting this true.

It does seem like a bit of extra complexity. I wonder if some single global flag would be better than adding/removing from a bunch of the APIs, where we look to that one single flag on how to behave in the absence of any other clues.

Reply:
I like the idea of such a global setting. I implemented a prototype of this which I will push in a moment. Let's see how this can play out.